handlers/delivery-problem-credit-processor/src/main/scala/com/gu/deliveryproblemcreditprocessor/DeliveryCreditProcessor.scala

```diff
@@ -74,14 +82,18 @@ object DeliveryCreditProcessor extends Logging {
     } yield creditResults.flatten
   }
 
-  def gatherCreditResults(processResult: ProcessResult[DeliveryCreditResult]): Task[List[DeliveryCreditResult]] =
+  def gatherCreditResults(processResults: List[ProcessResult[DeliveryCreditResult]]): Task[List[DeliveryCreditResult]] =
     for {
-      _ <- Task.effect(ProcessResult.log(processResult))
-      _ <- Task.effect(processResult.overallFailure).flatMap {
-        case None => Task.succeed(())
-        case Some(e) => Task.fail(new RuntimeException(e.reason))
+      _ <- Task.effect(ProcessResult.log(processResults))
+      _ <- Task.effect(processResults).map { pr =>
+        pr.map {
+          _.overallFailure match {
+            case None => Task.succeed(())
+            case Some(e) => Task.fail(new RuntimeException(e.reason))
+          }
+        }
```

Comment:
I'm not sure this will do what you expect?
the type would be Task[List[Task[Unit]]] wouldn't it? but the inner Task might be failed, and previously we would have short circuited the for comp, but now we are discarding the value.
I think we might need something more akin to 
```suggestion
      pr <- Task.effect(processResults)
      _ <-  pr.traverse {
          _.overallFailure match {
            case None => Task.succeed(())
            case Some(e) => Task.fail(new RuntimeException(e.reason))
          }
```

Reply:
I think we'll have to pair on this i dont understand Zio enough to get that to work